ra2ce/analysis/damages/damage_functions/damage_function_road_type_lane.py

```diff
@@ -18,56 +18,38 @@
     You should have received a copy of the GNU General Public License
     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """
-
-import logging
+from dataclasses import dataclass
 from pathlib import Path
 
 import pandas as pd
 
-from ra2ce.analysis.damages.damage.damage_fraction_uniform import DamageFractionUniform
-from ra2ce.analysis.damages.damage.max_damage import MaxDamageByRoadTypeByLane
+from ra2ce.analysis.damages.damage_functions.damage_fraction_uniform import (
+    DamageFractionUniform,
+)
+from ra2ce.analysis.damages.damage_functions.max_damage import MaxDamage
 
 
+@dataclass(kw_only=True)
 class DamageFunctionByRoadTypeByLane:
     """
     A damage function that has different max damages per road type, but a uniform damage_fraction curve
 
-
     The attributes need to be of the type:
-    self.max_damage (MaxDamage_byRoadType_byLane)
-    self.damage_fraction (DamageFractionHazardSeverityUniform)
-
+    self.max_damage (MaxDamage)
+    self.damage_fraction (DamageFractionUniform)
+    name (str)
     """
 
-    def __init__(
-        self,
-        max_damage: MaxDamageByRoadTypeByLane = None,
-        damage_fraction: DamageFractionUniform = None,
-        name: str = "",
-        hazard: str = "flood",
-        type: str = "depth_damage",
-        infra_type: str = "road",
-    ):
-        # Construct using the parent class __init__
-        self.name = name
-        self.hazard = hazard
-        self.type = type
-        self.infra_type = infra_type
-        self.max_damage = max_damage  # Should be a MaxDamage object
-        self.damage_fraction = (
-            damage_fraction  # Should be a DamageFractionHazardSeverity object
-        )
-        self.prefix = None  # Should be two characters long at maximum
-
-    def set_prefix(self):
-        self.prefix = self.name[0:2]
-        logging.info(
-            "The prefix: '{}' refers to curve name '{}' in the results".format(
-                self.prefix, self.name
-            )
-        )
+    max_damage: MaxDamage = None
+    damage_fraction: DamageFractionUniform = None
+    name: str = None
+
+    @property
+    def prefix(self) -> str:
+        return self.name[0:2]
```

Comment:
this, by definition, will crash. As `name: str = None`, therefore you need to add a check. In addition I advise setting the `str` default value to `""`:
```suggestion
    name: str = ""

    @property
    def prefix(self) -> str:
        return self.name[0:2] if len(self.name)>2 else self.name
```

This will return an empty string or any available character when the name is composed by no more than 2 characters.

Reply:
Better!